extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/RadiusConnectionService.java

```diff
@@ -129,6 +133,18 @@ private RadiusAuthenticator setupRadiusAuthenticator(RadiusClient radiusClient)
         if (radAuth == null)
             throw new GuacamoleException("Could not get a valid RadiusAuthenticator for specified protocol: " + confService.getRadiusAuthProtocol());
 
+        // For MSCHAPv1/2, we need MD4 support
+        if (radAuth instanceof MSCHAPv1Authenticator
+                || radAuth instanceof MSCHAPv2Authenticator) {
```

Comment:
Rather than checking `instanceof` everywhere, and thus relying on the internals of JRadius, it would be better if `getRadiusAuthProtocol()` leveraged an `enum` that could be checked independently of whatever JRadius decides to return. Such an `enum` could provide a method for obtaining a corresponding instance of `RadiusAuthenticator` given a `RadiusClient`.

Rechecking `getRadiusAuthProtocol()`, it is somewhat scary that the raw string value is passed straight to JRadius. I think there should be some validation on the value on the Guacamole side, and a decoupling from the underlying RADIUS library.

Reply:
Okay, I'll work up a new property type and `enum` for this.

Reply:
This should be implemented as requested.